pkg/controller/datavolume-controller.go

```diff
@@ -1567,10 +1608,11 @@ func (r *DatavolumeReconciler) storageClassCSIDriverExists(storageClassName *str
 func (r *DatavolumeReconciler) getSnapshotClassForSmartClone(dataVolume *cdiv1.DataVolume, targetStorageSpec *corev1.PersistentVolumeClaimSpec) (string, error) {
 	log := r.log.WithName("getSnapshotClassForSmartClone").V(3)
 	// Check if relevant CRDs are available
-	if !IsCsiCrdsDeployed(r.extClientSet) {
+	if !IsCsiCrdsDeployed(r.client, r.log) {
 		log.Info("Missing CSI snapshotter CRDs, falling back to host assisted clone")
 		return "", nil
 	}
+	r.sccs.StartController()
```

Comment:
This seems a little too hidden in this function, how about here:  https://github.com/kubevirt/containerized-data-importer/blob/f68376fe09b2e72937e9b42712700e0f78404dc0/pkg/controller/datavolume-controller.go#L482

Check if strategy is smart clone and start controller if so.  In this location (main reconcile func) it is more explicit, no weird side effects

Reply:
I mean if we do manage to attempt to start the smart clone controller, it will fail when we add the watches in the controller because the CRDs don't exist. but you make a good point that we won't pick that strategy without checking if the CRDs exist. I already fixed it btw.